### PR TITLE
Updated README with up-to-date info and calls to action

### PR DESCRIPTION
**Pull request by akosyakov**
The resulted doc is here: https://github.com/theia-ide/theia/blob/2e7466f931a7af33baff039aae26f05a189f7528/README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,74 @@
...
+# Theia - Build Cloud & Desktop IDEs with modern web tech.
 [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square)](https://github.com/theia-ide/theia/labels/help%20wanted)
 [![Gitter](https://img.shields.io/badge/chat-on%20gitter-blue.svg)](https://gitter.im/theia-ide/theia)
 [![Build Status](https://travis-ci.org/theia-ide/theia.svg?branch=master)](https://travis-ci.org/theia-ide/theia)
 [![Build status](https://ci.appveyor.com/api/projects/status/02s4d40orokl3njl/branch/master?svg=true)](https://ci.appveyor.com/project/kittaakos/theia/branch/master)
 [![Open questions](https://img.shields.io/badge/Open-questions-pink.svg?style=flat-square)](https://github.com/theia-ide/theia/labels/question)
 [![Open bugs](https://img.shields.io/badge/Open-bugs-red.svg?style=flat-square)](https://github.com/theia-ide/theia/labels/bug)
 
-Theia is a cloud & desktop IDE framework implemented in TypeScript.
+Theia is an extensible platform to develop a full-fledged multi-language Cloud & Desktop IDE-like products with state-of-the-art web technologies.
```
Comment (hexa00) [inline README.md]: Theia is an extensible platform to develop full-fledged multi-language Cloud & Desktop IDE-like products with state-of-the-art web technologies.

(removed the `a` in `develop a full-fledged`)

File: README.md [hunk 2]
```diff
@@ -1,24 +1,74 @@
...
+- [**Developing an extension**](doc/Authoring_Extensions.md)
+  - [**Frontend-backend communications**](doc/Internals.md#backendfrontend)
+  - [**Events**](doc/Internals.md#events)
+  - [**Commands**](doc/Commands_Keybindings.md)
+  - [**Preferences**](doc/Preferences.md)
+
+## Contributing
+
+Read below to learn how to take part in improving Theia:
+- Fork the repository and [run examples from source](doc/Developing.md#quick-start)
```
Comment (hexa00) [inline README.md]: run `the` examples from source

File: README.md [hunk 3]
```diff
@@ -1,24 +1,74 @@
...
+- Get familiar with [the development workflow](doc/Developing.md), [Coding Guidlines](https://github.com/theia-ide/theia/wiki/Coding-Guidelines), [Code of Conduct](CODE_OF_CONDUCT.md) and [learn how to sign your work](CONTRIBUTING.md#sign-your-work)
+- Find an issue to work on and submit a pull request
+  - First time contirbuting to open source? Pick a [good first issue](https://github.com/theia-ide/theia/labels/good%20first%20issue) to get you familiar with GitHub contribution process.
```
Comment (hexa00) [inline README.md]: typo `contributing`

Comment (hexa00) [inline README.md]: there's no match for good first issue.. makes it a bit weird ?

**Comment by hexa00** [comment]
Otherwise it's very nice, thanks!

**Comment by akosyakov** [comment]
I’ve updated it

**Pull request merged**